Dalamud/Interface/Components/ImGuiComponents.IconButton.cs

```diff
@@ -52,6 +52,7 @@ public static bool IconButton(string iconText)
     /// <param name="activeColor">The color of the button when active.</param>
     /// <param name="hoveredColor">The color of the button when hovered.</param>
     /// <returns>Indicator if button is clicked.</returns>
+    [Obsolete("This version does not set an id for ImGui to use, recommended to use overload with Id field.", false)]
```

Comment:
`PushId` is a thing as well, are you sure you want to require double ID work?

Reply:
Can you elaborate on what you mean double Id work?

Since the Id's are an internal construct, people that are using PushId already, can either remove it, or can put whatever additional id they want, this comes at no real cost to usability.

Reply:
As a quick pseudocode example:

```cs
foreach (var thing in myThings) {
  _ = using ImRaii.PushId($"thingList-{thing.Id}");
  
  ImGui.Text($"Thing: {thing.Name}");

  if (IconButton(FontAwesome.MagnifyingGlass)) {
    thing.PrintInfo();
  }

  if (IconButton(FontAwesome.Trashcan)) {
    myThings.Remove(thing);
  }
}
```

In this case, the button already gets a unique ID and this change would require double-declaration of an ID where it's not strictly necessary.

Reply:
I feel this is a minority of cases, and this'll solve problems, but add a minor hassle to a minority of devs.

Folks using the code as you describe above would indeed likely just copy paste their Id into the IconButton field, but that comes at no harm.

Reply:
> minority of cases

More like almost 300: https://github.com/search?q=%22ImGuiComponents.IconButton%28FontAwesomeIcon%22&type=code&ref=advsearch&p=1

Most of the times you either way either want to use unique button icon in your UI or you're displaying in cycle and use PushId/PopId anyway. Cases where you'd use same icon outside of the loop aren't so very common I believe.